### PR TITLE
Doc fixes for sc-telemetry & API struct rename

### PR DESCRIPTION
**Pull request by cecton**
Related to #7463

polkadot companion: https://github.com/paritytech/polkadot/pull/2348

### CODE REVIEW

**Comment by cecton** [comment]
@dvdplm Thanks a lot for the review! You can continue :pray:

**Comment by cecton** [comment]
> …but I think the booleans should be flipped so we'd have logger.with_log_reloading(self.log_reloading).
>
> The origin of these funny booleans isn't this PR, but maybe we can avoid extending the awkwardness and have to do one bool flip here, and then another one in GlobalLoggerBuilder::with_log_reloading(). I'd rename GlobalLoggerBuilder.disable_log_reloading to just log_reloading: bool and work backwards from there.

It all comes down to the CLI. In the cli we have a parameter `--disable-log-reloading`. It's good as it is because the normal behavior is to have the log reloading and then the user have to insert `--disable-log-reloading` if they want to deactivate it.

On the builder though it makes more sense to have positive function. That is why I used `with_log_reloading` even though the default is to have it enabled.

I'm not sure why I kept `disable_log_reloading` internally :thinking: this could be changed to just `log_reloading` I guess.

**Comment by cecton** [comment]
@dvdplm I just realized now that the macro that was supposed to disable the log reloading was actually enabling the log reloading :laughing: 

So I fixed that at the same time.

File: client/service/src/config.rs [hunk 1]
```diff
@@ -213,11 +213,8 @@ impl Configuration {
 			return None;
 		}
 
-		match self.telemetry_endpoints.as_ref() {
-			// Don't initialise telemetry if `telemetry_endpoints` == Some([])
-			Some(endpoints) if !endpoints.is_empty() => Some(endpoints),
-			_ => None,
-		}
+		// Don't initialise telemetry if `telemetry_endpoints` == Some([])
+		self.telemetry_endpoints.as_ref().filter(|x| !x.is_empty())
```
Comment (cecton) [inline client/service/src/config.rs:217]: Ahh there it is @dvdplm . See! I knew I did it. It was just in another PR

**Comment by cecton** [comment]
This is also ready for review

**Review by dvdplm — APPROVED** [review]
lovely, ty.

File: client/tracing/src/logging/directives.rs [hunk 2]
```diff
@@ -114,7 +114,7 @@ pub(crate) fn set_reload_handle(handle: Handle<EnvFilter, SCSubscriber>) {
 	let _ = FILTER_RELOAD_HANDLE.set(handle);
 }
 
-// The layered Subscriber as built up in `init_logger()`.
+// The layered Subscriber as built up in `GlobalLogger::init()`.
```
Comment (bkchr) [inline client/tracing/src/logging/directives.rs:117]:
```suggestion
// The layered Subscriber as built up in `LoggerBuilder::init()`.
```

**Review by bkchr — APPROVED** [review]

**Comment by cecton** [comment]
bot merge

**Pull request merged**